src/Controls/tests/DeviceTests/Elements/Layout/LayoutTests.Android.cs

```diff
@@ -1,18 +1,61 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
+using Android.Graphics.Drawables;
+using AndroidX.AppCompat.Widget;
+using AndroidX.Core.Widget;
 using Microsoft.Maui.Controls;
-using Microsoft.Maui.Graphics;
+using Microsoft.Maui.DeviceTests.Stubs;
 using Microsoft.Maui.Handlers;
+using Microsoft.Maui.Hosting;
 using Microsoft.Maui.Platform;
 using Xunit;
 
 namespace Microsoft.Maui.DeviceTests
 {
 	public partial class LayoutTests
 	{
+		[Fact, Category(TestCategory.Layout)]
+		public async Task NestedButtonHasExpectedIconPosition()
+		{
+			EnsureHandlerCreated(builder =>
+			{
+				builder.ConfigureMauiHandlers(handlers =>
+				{
+					handlers.AddHandler<Page, PageHandler>();
+					handlers.AddHandler<Window, WindowHandlerStub>();
+					handlers.AddHandler<Layout, LayoutHandler>();
+					handlers.AddHandler<Button, ButtonHandler>();
+				});
+			});
+
+			var button = new Button()
+			{
+				Text = "Hello",
+				ImageSource = "red.png",
+				ContentLayout = new Button.ButtonContentLayout(Button.ButtonContentLayout.ImagePosition.Right, 8)
+			};
+
+			var page = new ContentPage()
+			{
+				Content = new HorizontalStackLayout()
+				{
+					button
+				}
+			};
+
+			await AttachAndRun(page, (_) =>
```

Comment:
Change this to CreateHandlerAndAddToWindow

Reply:
Updated